src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -75,16 +93,68 @@ void tryCompileDeepFluentCalls(int depth)
 }");
 
                 var source = builder.ToString();
-
-                var thread = new System.Threading.Thread(() =>
+                RunInThread(() =>
                 {
                     var options = new CSharpCompilationOptions(outputKind: OutputKind.DynamicallyLinkedLibrary, concurrentBuild: false);
                     var compilation = CreateCompilation(source, options: options);
                     compilation.VerifyDiagnostics();
                     compilation.EmitToArray();
-                }, 0);
-                thread.Start();
-                thread.Join();
+                });
+            }
+        }
+
+        [Fact]
+        [WorkItem(33909, "https://github.com/dotnet/roslyn/issues/33909")]
+        public void DeeplyNestedGeneric()
+        {
+            int nestingLevel = 500;
```

Comment:
>int nestingLevel = 500; [](start = 12, length = 23)

I'm surprised this is not per-platform. Did you just pick a level that works everywhere? If so, I'm not sure that is the best for detecting changes. 
Might it be possible to regress in RELEASE without regressing in DEBUG for example?

Reply:
I will tune this a biit. 

---
In reply to: [263977314](https://github.com/dotnet/roslyn/pull/33985#discussion_r263977314) [](ancestors = 263977314)

Reply:
Consider adding some code to detect and print out the current limit (as we do in `OverflowOnFluentCall` above)